sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +765,73 @@
             return monoError(logger, ex);
         }
     }
+
+    /**
+     * Initiates a search and returns a {@link PhoneNumberSearch} usable by other functions
+     * This function returns a Long Running Operation poller that allows you to 
+     * wait indefinitely until the operation is complete.
+     * 
+     * @param options A {@link CreateSearchOptions} with the search options
+     * @param lroDuration The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the search result
+     */
+    public PollerFlux<PhoneNumberSearch, PhoneNumberSearch> beginCreateSearch(
+        CreateSearchOptions options, Duration lroDuration) {
+        final Duration interval = lroDuration != null ? lroDuration : DEFAULT_POLL_INTERVAL;
+        return new PollerFlux<PhoneNumberSearch, PhoneNumberSearch>(interval,
+            createSearchActivationOperation(options),
+            createSearchPollOperation(),
+            (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            createSearchFetchResultOperation());
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PhoneNumberSearch>> createSearchActivationOperation(CreateSearchOptions options) {
+        return (pollingContext) -> {
+            Mono<PhoneNumberSearch> response = createSearch(options).flatMap(createSearchResponse ->
+            {
+                String searchId = createSearchResponse.getSearchId();
+                PhoneNumberSearch phoneNumberSearch = new PhoneNumberSearch().setSearchId(searchId);
+                return Mono.just(phoneNumberSearch);
+            });
+            return response;
+        };
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PollResponse<PhoneNumberSearch>>> createSearchPollOperation() {
+        return pollingContext ->
+            getSearchById(pollingContext.getActivationResponse().getValue().getSearchId())
+                .flatMap(getSearchResponse -> {
+                    if (getSearchResponse.getStatus().equals(SearchStatus.SUCCESS)) {
+                        return Mono.just(new PollResponse<>(
```

Comment:
Can we clean this up a bit, get the status in a variable and set do the same this for Expired, cancelled, etc..

Reply:
I see this is still not fixed